Makefile

```diff
@@ -1,20 +1,28 @@
-#Top-level Makefile for AudioReach kernel modules
+# Default to environment variable or fallback path for the kernel source
+KERNEL_SRC ?= /local/mnt/workspace/kernel_mainline/kernel
 
+# A list of subdirectories containing Kbuild Makefiles
 SUBDIRS := audioreach-driver
 
-.PHONY: all clean modules_install
-
+# The default target builds all modules in the subdirectories
 all:
 	@for dir in $(SUBDIRS); do \
-		$(MAKE) -C $$dir all; \
+		echo "Building modules in $$dir..."; \
+		$(MAKE) -C $(KERNEL_SRC) M=$(shell pwd)/$$dir modules; \
```

Comment:
The issue is different. You should be using `$(CURDIR) inside the subdir's Makefile, not `$(PWD)`. Your code works, but it is very clumsy.